### ISSUE TITLE
eslint ignores symbolic links to files when traversing directories

### ISSUE THREAD

**Issue by rtpg** [eslint/eslint#13615]
In the following code we have an `.isFile` check on a `Dirent` object

https://github.com/eslint/eslint/blob/091e52ae1ca408f3e668f394c14d214c9ce806e6/lib/cli-engine/file-enumerator.js#L429

However [the node documentation](https://nodejs.org/dist/latest-v12.x/docs/api/fs.html#fs_dirent_isfile) says the `isFile` check is for if the file is a _regular file_, and returns false on symbolic links.

This makes it hard to run eslint against sandboxed bazel files, for example (since the sandboxing happens through symlinks)

Not 100% sure what's best here but "not a directory" feels like a nicer check here than "is a regular file". Might be a bit impossible to resolve.

**Comment by mdjermanovic** [comment]
Hi @rtpg, thanks for the issue!

I think this is similar to #13551, but for files. Looks like a bug to me.

Until we verify and possibly fix this, a workaround (though an inconvenient one if you have multiple symlinked files) is to explicitly list those files as command-line arguments. For example:

```bash
 eslint src src/file1.js 
```

That way, the `src/file1.js ` path will be checked [here](https://github.com/eslint/eslint/blob/091e52ae1ca408f3e668f394c14d214c9ce806e6/lib/cli-engine/file-enumerator.js#L336). `statSafeSync(absolutePath).isFile()` follows links and returns `true` if it is a file.

**Comment by mdjermanovic** [comment]
Reopening as this is most likely a bug.

**Comment by blindspeed90** [comment]
I'm not a Linux expert but glob expansion didn't recurse and work for extensions (like "/myrootsrc/**/*.js").  Instead I used "find" for "js" files only, in case anyone else wants a workaround (and yes it can create a long cmdline, and I don't have spaces in paths):

ROOT_DIR=/myrootsrc
# TODO: Hack 7.x ignoring symbolic links by finding the files ourselves and passing them in
ROOT_DIR=$(find $ROOT_DIR -name "*.js")
auditjs $AUDIT_OPTS $ROOT_DIR

**Comment by mrl5** [comment]
I'm interested if there are any plans to fix this

**Comment by mdjermanovic** [comment]
Hi @mrl5! This will be fixed by #14126.

**Comment by bidoubiwa** [comment]
Hey! This introduced a breaking change for me. 
```
Oops! Something went wrong! :(

ESLint: 7.21.0

Error: ELOOP: too many symbolic links encountered,
```
